apps/server/src/modules/deletion/uc/deletion-request.uc.ts

```diff
@@ -0,0 +1,269 @@
+	): Promise<{ requestId: EntityId; deletionPlannedAt: Date }> {
+		const result = await this.deletionRequestService.createDeletionRequest(
+			deletionRequest.targetRef.targetRefId,
+			deletionRequest.targetRef.targetRefDoamin,
+			deletionRequest.deleteInMinutes
+		);
+
+		return result;
+	}
+
+	async executeDeletionRequests(options?: IFindOptions<DeletionRequest>): Promise<void> {
+		const deletionRequestToExecution: DeletionRequest[] = await this.deletionRequestService.findAllItemsToExecute(
+			options
+		);
+
+		for (const req of deletionRequestToExecution) {
+			// eslint-disable-next-line no-await-in-loop
+			await this.executeDeletionRequest(req);
+		}
+	}
+
+	async findById(deletionRequestId: EntityId): Promise<DeletionRequestLog> {
+		const deletionRequest: DeletionRequest = await this.deletionRequestService.findById(deletionRequestId);
+		let response: DeletionRequestLog = {
+			targetRef: {
+				targetRefDoamin: deletionRequest.targetRefDomain,
+				targetRefId: deletionRequest.targetRefId,
+			},
+			deletionPlannedAt: deletionRequest.deleteAfter,
+		};
+
+		if (deletionRequest.status === DeletionStatusModel.SUCCESS) {
+			const deletionLog: DeletionLog[] = await this.deletionLogService.findByDeletionRequestId(deletionRequestId);
+			const deletionLogStatistic: DeletionLogStatistic[] = deletionLog.map((log) => {
+				return {
+					domain: log.domain,
+					modifiedCount: log.modifiedCount,
+					deletedCount: log.deletedCount,
+				};
+			});
+			response = { ...response, statistics: deletionLogStatistic };
+		}
+
+		return response;
+	}
+
+	async deleteDeletionRequestById(deletionRequestId: EntityId): Promise<void> {
+		await this.deletionRequestService.deleteById(deletionRequestId);
+	}
+
+	private async executeDeletionRequest(deletionRequest: DeletionRequest): Promise<void> {
+		await Promise.all([
+			this.accountService.deleteByUserId(deletionRequest.targetRefId),
+			this.classService.deleteUserDataFromClasses(deletionRequest.targetRefId),
+			this.courseGroupService.deleteUserDataFromCourseGroup(deletionRequest.targetRefId),
+			this.courseService.deleteUserDataFromCourse(deletionRequest.targetRefId),
+			this.filesService.markFilesOwnedByUserForDeletion(deletionRequest.targetRefId),
+			this.filesService.removeUserPermissionsToAnyFiles(deletionRequest.targetRefId),
+			this.lessonService.deleteUserDataFromLessons(deletionRequest.targetRefId),
+			this.pseudonymService.deleteByUserId(deletionRequest.targetRefId),
+			this.teamService.deleteUserDataFromTeams(deletionRequest.targetRefId),
+			this.userService.deleteUser(deletionRequest.targetRefId),
+			this.removeUserFromRocketChatByUserId(deletionRequest.targetRefId),
+		])
+			.then(
+				async ([
+					,
+					classesUpdated,
+					courseGroupUpdated,
+					courseUpdated,
+					fileDeleted,
+					filesPermisionUpdated,
+					lessonUpdated,
+					pseudonymDeleted,
+					teamsUpdated,
+					userDeleted,
+					rocketChatUserDeleted,
+				]) => {
```

Comment:
consider splitting it to separate methods, at least move this logs to diffrent method